metals/src/main/scala/scala/meta/internal/metals/BuildTargets.scala

```diff
@@ -211,6 +215,31 @@ final class BuildTargets() {
     }
   }
 
+  def inverseSourcesBsp(
+      source: AbsolutePath
+  )(implicit ec: ExecutionContext): Future[Option[BuildTargetIdentifier]] = {
+    inverseSources(source) match {
+      case None =>
+        val identifier = new TextDocumentIdentifier(
+          source.toTextDocumentIdentifier.getUri()
+        )
+        val params = new InverseSourcesParams(identifier)
+        val connections =
+          data.fromIterators(_.targetToConnection.values.toIterator).distinct
+        val queries = connections.map { connection =>
+          connection
+            .buildTargetInverseSources(params)
+            .map(_.getTargets.asScala.toList)
+        }
+        Future.sequence(queries).map { results =>
+          val target = results.flatten.maxByOption(buildTargetsOrder)
+          target
```

Comment:
Can we register the result to https://github.com/scalameta/metals/blob/934ac115a3469f5585f284023da804f04bfa2877/metals/src/main/scala/scala/meta/internal/metals/BuildTargets.scala#L41-L42 to reuse the result as cache in the next request?

Reply:
Thanks for the suggestion! I made the change. Is that what you had in mind?